app/src/main/java/com/github/swent/echo/data/room/entity/EventRoom.kt

```diff
@@ -0,0 +1,79 @@
+package com.github.swent.echo.data.room.entity
+
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.Index
+import androidx.room.Junction
+import androidx.room.PrimaryKey
+import androidx.room.Relation
+import com.github.swent.echo.data.model.Event
+import java.time.ZonedDateTime
+
+@Entity
+data class EventRoom(
+    @PrimaryKey val eventId: String,
+    @Embedded val creator: EventCreatorRoom,
```

Comment:
If I understand correctly how `@Embedded` works, this will add the fields of `EventCreatorRoom` to the Event table. I'm not sure if this is actually what we want. The event creator should be a subset of the attributes of `userProfile` and then it could be a DB relation between `UserProfile` and `Event`. Otherwise changes to the `UserProfile` won't be visible on the events.

Maybe [views](https://developer.android.com/training/data-storage/room/creating-views) could help us implement this properly? However they only allow to do "SELECT" queries. 
Also the downside of this approach would be that we'd either have a situation where `userProfile`s would be incomplete in the local database (i.e. only the attributes part of `CreatorProfile` have been loaded from the remote). Or we'd need to fetch the complete the compete `UserProfile`s as soon as we have an Event referencing it (not what we want either).

What do you think makes the most sense here? Would the "incomplete `UserProfile`s" approach make sense to you?

Reply:
Yeah, you're right, it's not the best solution. My thinking was that the name of the user won't change often. Therefore, I thought it's acceptable that in some rare cases, it might be out of sync with the real user profile name. Also, we are using the local data source only for caching and not as a source of truth. We could add some mechanism in the UI (e.g. pull down to refresh) to update/sync/refresh the data...
Otherwise, we could just not offer the possibility to the user to change his name... this we might also discuss with the team.

Reply:
Agree that that is a low priority thing and as you say it's not the source of truth it's acceptable to be out of sync. Offering a pull-to-refresh to force-refresh is a good idea! We should look into this.
Treating the names (or similar attributes) as immutable however is generally a bad idea. Users might simply misspell it on profile creation and want to correct it or want to change names for a diverse range of reason (marriage, gender, identity, etc)...

Reply:
Yes, you're right, we should let the user the freedom to change their name. But, as you said, this is not high priority, so we should leave this for the future and keep this solution for now!

Reply:
I opened an issue #166 for the pull-to-refresh support.